packages/nimble-components/src/table/specs/cell-state-when-scrolling-hld.md

```diff
@@ -0,0 +1,111 @@
+
+**Note / Behavior limitation in Firefox**: You can only select a single cell's text. Once you cross into another cell/column the selection gets cleared. The selection also resets if you start text selection from outside the table, then move your cursor within the table viewport (to the rows/cells) to try to select more text there. There doesn't seem to be anything for us to do about this, it's probably related to the table's use of Shadow DOM.
+
+**Possible Implementation Plan: Detecting Table Text Selection**
+
+To check if the table contains selected text:  
+Get `window.getSelection()`. If null or `rangeCount === 0`, no text is selected. Otherwise, for each `Range`:
+
+-   Check if `startContainer`/`endContainer` is the `nimble-table`
+-   Else, do the `compareBoundaryPoints` check (in a `try/catch`)
+-   Else, check if `startContainer`/`endContainer` has the `nimble-table` as an ancestor
+
+If any of those `Range` checks succeed, remove that `Range` from the selection.  
+_Limitations:_
+
+-   If text is selected in the table but outside rows (e.g. column header text), it will also be cleared when the user scrolls. (There's not a good way to differentiate the text location that works in each browser.)
+-   (Safari only) If text selection is partially in the table and partially before/ after it, there isn't any way for us to detect that case. So Safari will still have text incorrectly selected after the scroll in that case.
+
+_API:_  
+This logic will be in the Nimble Virtualizer class, called from `handleVirtualizerChange()` (called when the user scrolls). It will apply to all Nimble tables / all column types, without an opt-out option.
+
+**Conclusion**: Since the code required to detect table selection is problematic (`try/catch`) and complex, doesn't work fully in Safari, and is potentially fragile, we currently don't plan to try and detect table text selection.
+
+### Disabling Table Text Selection
+
+We could add CSS to our table text cells (`user-select: none`) to prevent text selection. This would also eliminate the concern about text selection remaining after a scroll.
+
+We decided against this because there's valid use cases for copying text out of a table cell, and the selection+scrolling issue doesn't seem like a sufficient reason to disable selection entirely.
+
+## Open Issues
+
+-   Decide between API Option 1 and Option 2 for blur-ing focused cell controls.
+-   Should we have a default, generic implementation of `onBeforeFocusedCellRecycled`? (It would look for known Nimble control types in the cell, and call the appropriate API on them, `blur()` or `open = false`)
+    -   Current we **do not** plan to have a default implementation. We think it's better for the column implementations to handle this, as they're the ones declaring the editable/blur-able controls in their templates.
```

Comment:
+1

Reply:
Took out this open issue. Added a section in Alternate Implementations about the idea of having this default logic in the blur handler, but that we decided against it.